src/Salad.java

```diff
@@ -0,0 +1,73 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+
+public class Salad {
+    private String name;
+    ArrayList<Vegetable> ingredients;
+    private int saladCalories = 0;
+
+    Salad(String name, Vegetable... ingredient) {
+        this.name = name;
+        ingredients = new ArrayList<Vegetable>();
+        for (Vegetable specificVeg : ingredient) {
+            this.saladCalories += specificVeg.getVegetableCalories();
+            this.ingredients.add(specificVeg);
+        }
+    }
+
+    public String getSaladName() {
+        return name;
+    }
+
+    public Salad setSaladName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    public int getSaladCalories() {
+        return saladCalories;
+    }
+
+    public void addIngredientToSalad(Vegetable ing) {
+        System.out.println("New ingredient is added.");
+        this.ingredients.add(ing);
+        this.saladCalories += ing.getVegetableCalories();
+    }
+
+    public void removeIngredientFromSalad(String name) {
+        Iterator itr = ingredients.iterator();
+        while (itr.hasNext()) {
+            Vegetable veg = (Vegetable) itr.next();
+            String vegNameToBeRemoved = veg.getVegetableName();
+            if (vegNameToBeRemoved.equals(name)) {
+                System.out.println("Ingredient is removed");
```

Comment:
what if itr.remove will fail? You've already print out message that it is removed but it may fail. Print the result after operation completed.